Ansible concepts: add ref to Getting started with EE

Andersson007 opened the pull request:
Ansible concepts:
- add ref to Getting started with EE
- improve readability

Andersson007 committed:
Ansible concepts: add ref to Getting started with EE

Andersson007 committed:
replace anchor with URL

oraNod commented on docs/docsite/rst/shared_snippets/basic_concepts.txt:
```diff
@@ -47,23 +60,36 @@ A special form of a Task, that only executes when notified by a previous task wh
+Collection resources can be used independently and discretely from each other.
 
 
 AAP
```
How about if we remove mentions of AAP here while we're at it? I'm not sure how well this qualifies as an "Ansible concept".

Andersson007 replied:
Done, i think the ref on the docs.ansible.com (already there) is enough.
PTAL

Andersson007 committed:
Change suggested

gotmax23 commented on docs/docsite/rst/shared_snippets/basic_concepts.txt:
```diff
@@ -1,42 +1,55 @@
+
 You can use any computer that meets the software requirements as a control node - laptops, shared desktops, and servers can all run Ansible.
+
+You can also run Ansible in containers known as `execution envoronments <https://docs.ansible.com/ansible/devel/getting_started_ee/index.html>`_.
```
Shouldn't this use a `:ref:`?

oraNod replied:
I think because it's currently only on the `devel` branch it needs this format. We'll also need to update this line again if we change the location of the execution environments getting started.

Andersson007 replied:
sure, let's do it later

oraNod commented on docs/docsite/rst/shared_snippets/basic_concepts.txt:
```diff
@@ -1,42 +1,55 @@
+
 You can use any computer that meets the software requirements as a control node - laptops, shared desktops, and servers can all run Ansible.
+
+You can also run Ansible in containers known as `execution envoronments <https://docs.ansible.com/ansible/devel/getting_started_ee/index.html>`_.
```
```suggestion
You can also run Ansible in containers known as `execution environments <https://docs.ansible.com/ansible/devel/getting_started_ee/index.html>`_.
```

Andersson007 committed:
Update docs/docsite/rst/shared_snippets/basic_concepts.txt

Co-authored-by: Don Naro <dnaro@redhat.com>

Andersson007 commented:
Feedback has been incorporated, PTAL

samccann commented:
It should be able to be a :ref:. If we update the intersphinx links before we merge this PR.

Andersson007 commented:
> It should be able to be a :ref:. If we update the intersphinx links before we merge this PR.

Cool, please let me know when i can change it to `:ref:`

samccann commented:
Good news/bad news here.  
Good news = don't need to do any updating locally of the intersphinx links. It just 'works'.
Bad news = can't add `ref` in this instance because the EE guide doesn't exist in the core docs build and thus it's an unknown reference.

Long story short - leave it as a url. It's unfortunate, but I don't know of any way around this conundrum.

samccann commented:
Okay viewing this as a comparison. The current page looks as follows:

![image](https://github.com/ansible/ansible-documentation/assets/30267855/f145258f-4db2-471b-9601-498d0afe5f72)

This PR looks as follows:
![image](https://github.com/ansible/ansible-documentation/assets/30267855/a3fe7c3b-137e-48ec-b8d1-707d90d2b335)

I disagree that this makes the content more readable. I think it just makes it take up unnecessary space. 

What do other folks think?

Andersson007 committed:
Incorporate feedback

Andersson007 commented:
@samccann i agree, it's too much.. thanks for pointing this out. I shouldn've generated the doc and taken a look first

I removed some spaces.

I also removed a line about necessity of a task in a play as it doesn't seem to be true:
```
[aklychko@localhost ees]$ ansible-navigator run playbook.yml -m stdout
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that
the implicit localhost does not match 'all'

PLAY [My play] *****************************************************************

PLAY RECAP *********************************************************************
[aklychko@localhost ees]$ 
[aklychko@localhost ees]$ 
[aklychko@localhost ees]$ cat playbook.yml 
- name: My play
  hosts: localhost
  gather_facts: false
```

PTAL

oraNod commented:
@Andersson007 A related thing, of which I'm a fan, are semantic line breaks. This is also referred to as ventilated prose and is a fantastic writing habit that is often overlooked. Basically it's "one sentence per line".

Most plain text formats (rst, md, adoc) render sentences on consecutive lines as paragraphs. Putting each sentence on its own line makes PR reviews a lot easier, allows you to swap and comment lines, easily spot mundane writing patterns. It also avoids text reflow issues that can be problematic if you're doing things like converting or replacing text during a pre-build process. I've found it also helps when reading directly from source.

Andersson007 commented:
thanks everyone for your great reviews!
